test_build_tag_push.py

```diff
@@ -0,0 +1,86 @@
+import unittest
+import build_tag_push
+import os
+import subprocess
+import argparse
+
+# args_app_yml_path = '/Users/alyssakelley/Desktop/test_mvp/ymltest.yml'
+# args_docker_hub_name = 'alyssakelley'
+
+# Question: as we talked about the parses before, I then tried to put them
+# in there own function and use them later on, but I still had to use the
+# special unittest.main line below. Should I take all of this away and
+# then just use hard coded test cases (like the path and docker_hub_name
+# as I have commented out above)?
+
+
+parser = argparse.ArgumentParser(description='Usage: python3'
+                                 'build_tag_push.py <docker_hub_name'
+                                 '<app_yml_path')
+parser.add_argument('docker_hub_name', type=str, help='DockerHub namespace')
+
+parser.add_argument('app_yml_path', type=str,
+                    help='Repository path to MVP Studio file')
+args = parser.parse_args()
+
+
+class TestBuildTagPush(unittest.TestCase):
+
+    def test_parse_app_file(self):  # Success.
+
+        data_from_app_yml = build_tag_push.parse_app_file(args.app_yml_path)
+        self.assertEqual(data_from_app_yml['name'], 'build-setup-test')
+        self.assertEqual(data_from_app_yml['build'], 'run_my_test_build.sh')
+        self.assertEqual(data_from_app_yml['test'], 'test_the_test.sh')
+        self.assertEqual(data_from_app_yml['docker_context'],
+                         ['build', 'static', 'Dockerfile'])
+
+    def test_build_check_app(self):  # Fails.
+
+        data_from_app_yml = build_tag_push.parse_app_file(args.app_yml_path)
+
+        result = subprocess.check_call(['bash', data_from_app_yml['test']])
+
+        self.assertIsEqual(result, build_tag_push.build_check_app(
+            data_from_app_yml, args.app_yml_path, True))
+
+# Question: You meantioned doing a try: except: type of logic here, and with
+# the other functions below that are labeled with "CHANGE" next to the headers.
+# I have minimal experience with unittests and I don't know what I am expected
+# to really check to see if it is happening. Since I changed all of the
+# check_calls to call in my original file, does that mean I should do
+# check_calls here and see what value they return?
```

Comment:
We probably want to change your script to return *specific* exception sub-classes on different types of failures. So, for example, this line in `build_check_app()`

```
subprocess.call(['bash', data_from_app_yml['build']])
```

would become:

```
build_res = subprocess.call(['bash', data_from_app_yml['build']])
if build_res != 0:
    throw BuildFailedException('The build failed')
```

(where `BuildFailedException` is a class you'd write). Then to test you'd have something like the examples I have in the comment above.